hyprtrails: initiates build on nixos-unstable channel (flake based system) and doesn't build anyways.

### Nixpkgs version

- Unstable (25.05)

### Describe the bug

Installing hyprtrails from pkgs.hyprlandPlugins.hyprtrails initiates a build, I don't know why. And then the build fails unexpectedly saying that there is a bug in the builder script. I am using nixpkgs nixos-unstable. I am not using git packages for either Hyprland or hyprland plugins as stated. 

[Here](https://github.com/rachitve6h2g/NixDots) is my configuration files. 

### Steps to reproduce

Install hyprland plugins on 25.05 using home-manager hyprland module: 
```
wayland.windowManager.hyprland.plugins = with pkgs.hyprlandPlugins; [
    hyprtrails
    hyprexpo
    hypr-dynamic-cursors
];
```

### Expected behaviour

The plugin should just download and load, without building 

### Screenshots

_No response_

### Relevant log output

```console
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/gw0fziwnq2i4m0h9wzfci37hrjz1xf2m-source/hyprtrails
source root is hyprtrails
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
fixing cmake files...
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/dcg5kq7pyrn13swa7y1nlprfbhaxcm1c-hyprtrails-0.49.0/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/dcg5kq7pyrn13swa7y1nlprfbhaxcm1c-hyprtrails-0.49.0/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/dcg5kq7pyrn13swa7y1nlprfbhaxcm1c-hyprtrails-0.49.0/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/dcg5kq7pyrn13swa7y1nlprfbhaxcm1c-hyprtrails-0.49.0/share/doc/hyprtrails -DCMAKE_INSTALL_INFODIR=/nix/store/dcg5kq7pyrn13swa7y1nlprfbhaxcm1c-hyprtrails-0.49.0/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/dcg5kq7pyrn13swa7y1nlprfbhaxcm1c-hyprtrails-0.49.0/share/man -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/dcg5kq7pyrn13swa7y1nlprfbhaxcm1c-hyprtrails-0.49.0/include -DCMAKE_INSTALL_SBINDIR=/nix/store/dcg5kq7pyrn13swa7y1nlprfbhaxcm1c-hyprtrails-0.49.0/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/dcg5kq7pyrn13swa7y1nlprfbhaxcm1c-hyprtrails-0.49.0/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/dcg5kq7pyrn13swa7y1nlprfbhaxcm1c-hyprtrails-0.49.0/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/sxp88zqazm7cjaqxcr01zfryc7n7f15v-gcc-wrapper-14.2.1.20250322/bin/strip -DCMAKE_RANLIB=/nix/store/sxp88zqazm7cjaqxcr01zfryc7n7f15v-gcc-wrapper-14.2.1.20250322/bin/ranlib -DCMAKE_AR=/nix/store/sxp88zqazm7cjaqxcr01zfryc7n7f15v-gcc-wrapper-14.2.1.20250322/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/dcg5kq7pyrn13swa7y1nlprfbhaxcm1c-hyprtrails-0.49.0
-- The C compiler identification is GNU 14.2.1
-- The CXX compiler identification is GNU 14.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/sxp88zqazm7cjaqxcr01zfryc7n7f15v-gcc-wrapper-14.2.1.20250322/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/sxp88zqazm7cjaqxcr01zfryc7n7f15v-gcc-wrapper-14.2.1.20250322/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PkgConfig: /nix/store/2crk9xnq5x9v7yf0r2nwkgj8qsmxr4ly-pkg-config-wrapper-0.29.2/bin/pkg-config (found version "0.29.2")
-- Checking for modules 'hyprland;libdrm;libinput;libudev;pangocairo;pixman-1;wayland-server;xkbcommon'
--   Found hyprland, version 0.49.0
--   Found libdrm, version 2.4.124
--   Found libinput, version 1.27.1
--   Found libudev, version 257
--   Found pangocairo, version 1.56.3
--   Found pixman-1, version 0.44.2
--   Found wayland-server, version 1.23.1
--   Found xkbcommon, version 1.8.1
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
-- Configuring done (1.5s)
-- Generating done (0.0s)
[33mCMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_INSTALL_BINDIR
    CMAKE_INSTALL_DOCDIR
    CMAKE_INSTALL_INCLUDEDIR
    CMAKE_INSTALL_INFODIR
    CMAKE_INSTALL_LIBEXECDIR
    CMAKE_INSTALL_LOCALEDIR
    CMAKE_INSTALL_MANDIR
    CMAKE_INSTALL_SBINDIR
    CMAKE_POLICY_DEFAULT_CMP0025

[0m
-- Build files have been written to: /build/hyprtrails/build
cmake: enabled parallel building
cmake: enabled parallel installing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: -j4 SHELL=/nix/store/xy4jjgw87sbgwylm5kn047d9gkbhsr9x-bash-5.2p37/bin/bash
[ 20%] [32mBuilding CXX object CMakeFiles/hyprtrails.dir/main.cpp.o[0m
[ 40%] [32mBuilding CXX object CMakeFiles/hyprtrails.dir/CMakeFiles/3.31.6/CompilerIdCXX/CMakeCXXCompilerId.cpp.o[0m
[ 60%] [32mBuilding CXX object CMakeFiles/hyprtrails.dir/TrailPassElement.cpp.o[0m
[ 80%] [32mBuilding CXX object CMakeFiles/hyprtrails.dir/trail.cpp.o[0m
In file included from [01m[K/build/hyprtrails/trail.cpp:7[m[K:
[01m[K/build/hyprtrails/globals.hpp:8:5:[m[K [01;31m[Kerror: [m[K'[01m[KCShader[m[K' does not name a type; did you mean '[01m[KSShader[m[K'?
    8 |     [01;31m[KCShader[m[K          trailShader;
      |     [01;31m[K^~~~~~~[m[K
      |     [32m[KSShader[m[K
[01m[K/build/hyprtrails/trail.cpp:[m[K In destructor '[01m[Kvirtual CTrail::[01;32m[K~CTrail[m[K()[m[K':
[01m[K/build/hyprtrails/trail.cpp:43:36:[m[K [01;35m[Kwarning: [m[K'[01m[Kbool HyprlandAPI::[01;32m[KunregisterCallback[m[K(void*, Hyprutils::Memory::CSharedPointer<std::function<void(void*, SCallbackInfo&, std::any)> >)[m[K' is deprecated [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wdeprecated-declarations-Wdeprecated-declarations]8;;[m[K]
   43 |     [01;35m[KHyprlandAPI::unregisterCallback(PHANDLE, pTickCb)[m[K;
      |     [01;35m[K~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~[m[K
In file included from [01m[K/build/hyprtrails/trail.hpp:6[m[K,
                 from [01m[K/build/hyprtrails/trail.cpp:1[m[K:
[01m[K/nix/store/67kfpnkh6c9r3dmhdiqjsnf8gjrh6fy5-hyprland-0.49.0-dev/include/hyprland/src/plugins/PluginAPI.hpp:163:33:[m[K [01;36m[Knote: [m[Kdeclared here
  163 |     APICALL [[deprecated]] bool [01;36m[KunregisterCallback[m[K(HANDLE handle, SP<HOOK_CALLBACK_FN> fn);
      |                                 [01;36m[K^~~~~~~~~~~~~~~~~~[m[K
[01m[K/build/hyprtrails/trail.cpp:[m[K In member function '[01m[Kvoid CTrail::[01;32m[KrenderPass[m[K(PHLMONITOR, const float&)[m[K':
[01m[K/build/hyprtrails/trail.cpp:136:34:[m[K [01;31m[Kerror: [m[K'[01m[Kstruct SGlobalState[m[K' has no member named '[01m[KtrailShader[m[K'
  136 |     glUseProgram(g_pGlobalState->[01;31m[KtrailShader[m[K.program);
      |                                  [01;31m[K^~~~~~~~~~~[m[K
[01m[K/build/hyprtrails/trail.cpp:139:40:[m[K [01;31m[Kerror: [m[K'[01m[Kstruct SGlobalState[m[K' has no member named '[01m[KtrailShader[m[K'
  139 |     glUniformMatrix3fv(g_pGlobalState->[01;31m[KtrailShader[m[K.proj, 1, GL_TRUE, glMatrix.getMatrix().data());
      |                                        [01;31m[K^~~~~~~~~~~[m[K
[01m[K/build/hyprtrails/trail.cpp:249:91:[m[K [01;35m[Kwarning: [m[Knarrowing conversion of '[01m[K((PWINDOW.Hyprutils::Memory::CSharedPointer<CWindow>::operator->()->CWindow::m_realPosition.Hyprutils::Memory::CSharedPointer<Hyprutils::Animation::CGenericAnimatedVariable<Hyprutils::Math::Vector2D, SAnimationContext> >::operator->()->Hyprutils::Animation::CGenericAnimatedVariable<Hyprutils::Math::Vector2D, SAnimationContext>::value().Hyprutils::Math::Vector2D::x - pMonitor.Hyprutils::Memory::CSharedPointer<CMonitor>::operator->()->CMonitor::m_position.Hyprutils::Math::Vector2D::x) / pMonitor.Hyprutils::Memory::CSharedPointer<CMonitor>::operator->()->CMonitor::m_size.Hyprutils::Math::Vector2D::x)[m[K' from '[01m[Kdouble[m[K' to '[01m[Kfloat[m[K' [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/C_002b_002b-Dialect-Options.html#index-Wnarrowing-Wnarrowing]8;;[m[K]
  249 |     box thisboxopengl = box{[01;35m[K(PWINDOW->m_realPosition->value().x - pMonitor->m_position.x) / pMonitor->m_size.x[m[K,
      |                             [01;35m[K~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~[m[K
[01m[K/build/hyprtrails/trail.cpp:250:91:[m[K [01;35m[Kwarning: [m[Knarrowing conversion of '[01m[K((PWINDOW.Hyprutils::Memory::CSharedPointer<CWindow>::operator->()->CWindow::m_realPosition.Hyprutils::Memory::CSharedPointer<Hyprutils::Animation::CGenericAnimatedVariable<Hyprutils::Math::Vector2D, SAnimationContext> >::operator->()->Hyprutils::Animation::CGenericAnimatedVariable<Hyprutils::Math::Vector2D, SAnimationContext>::value().Hyprutils::Math::Vector2D::y - pMonitor.Hyprutils::Memory::CSharedPointer<CMonitor>::operator->()->CMonitor::m_position.Hyprutils::Math::Vector2D::y) / pMonitor.Hyprutils::Memory::CSharedPointer<CMonitor>::operator->()->CMonitor::m_size.Hyprutils::Math::Vector2D::y)[m[K' from '[01m[Kdouble[m[K' to '[01m[Kfloat[m[K' [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/C_002b_002b-Dialect-Options.html#index-Wnarrowing-Wnarrowing]8;;[m[K]
  250 |                             [01;35m[K(PWINDOW->m_realPosition->value().y - pMonitor->m_position.y) / pMonitor->m_size.y[m[K,
      |                             [01;35m[K~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~[m[K
[01m[K/build/hyprtrails/trail.cpp:251:99:[m[K [01;35m[Kwarning: [m[Knarrowing conversion of '[01m[K((PWINDOW.Hyprutils::Memory::CSharedPointer<CWindow>::operator->()->CWindow::m_realPosition.Hyprutils::Memory::CSharedPointer<Hyprutils::Animation::CGenericAnimatedVariable<Hyprutils::Math::Vector2D, SAnimationContext> >::operator->()->Hyprutils::Animation::CGenericAnimatedVariable<Hyprutils::Math::Vector2D, SAnimationContext>::value().Hyprutils::Math::Vector2D::x + PWINDOW.Hyprutils::Memory::CSharedPointer<CWindow>::operator->()->CWindow::m_realSize.Hyprutils::Memory::CSharedPointer<Hyprutils::Animation::CGenericAnimatedVariable<Hyprutils::Math::Vector2D, SAnimationContext> >::operator->()->Hyprutils::Animation::CGenericAnimatedVariable<Hyprutils::Math::Vector2D, SAnimationContext>::value().Hyprutils::Math::Vector2D::x) / pMonitor.Hyprutils::Memory::CSharedPointer<CMonitor>::operator->()->CMonitor::m_size.Hyprutils::Math::Vector2D::x)[m[K' from '[01m[Kdouble[m[K' to '[01m[Kfloat[m[K' [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/C_002b_002b-Dialect-Options.html#index-Wnarrowing-Wnarrowing]8;;[m[K]
  251 |                             [01;35m[K(PWINDOW->m_realPosition->value().x + PWINDOW->m_realSize->value().x) / pMonitor->m_size.x[m[K,
      |                             [01;35m[K~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~[m[K
[01m[K/build/hyprtrails/trail.cpp:252:99:[m[K [01;35m[Kwarning: [m[Knarrowing conversion of '[01m[K((PWINDOW.Hyprutils::Memory::CSharedPointer<CWindow>::operator->()->CWindow::m_realPosition.Hyprutils::Memory::CSharedPointer<Hyprutils::Animation::CGenericAnimatedVariable<Hyprutils::Math::Vector2D, SAnimationContext> >::operator->()->Hyprutils::Animation::CGenericAnimatedVariable<Hyprutils::Math::Vector2D, SAnimationContext>::value().Hyprutils::Math::Vector2D::y + PWINDOW.Hyprutils::Memory::CSharedPointer<CWindow>::operator->()->CWindow::m_realSize.Hyprutils::Memory::CSharedPointer<Hyprutils::Animation::CGenericAnimatedVariable<Hyprutils::Math::Vector2D, SAnimationContext> >::operator->()->Hyprutils::Animation::CGenericAnimatedVariable<Hyprutils::Math::Vector2D, SAnimationContext>::value().Hyprutils::Math::Vector2D::y) / pMonitor.Hyprutils::Memory::CSharedPointer<CMonitor>::operator->()->CMonitor::m_size.Hyprutils::Math::Vector2D::y)[m[K' from '[01m[Kdouble[m[K' to '[01m[Kfloat[m[K' [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/C_002b_002b-Dialect-Options.html#index-Wnarrowing-Wnarrowing]8;;[m[K]
  252 |                             [01;35m[K(PWINDOW->m_realPosition->value().y + PWINDOW->m_realSize->value().y) / pMonitor->m_size.y[m[K};
      |                             [01;35m[K~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~[m[K
[01m[K/build/hyprtrails/trail.cpp:253:33:[m[K [01;31m[Kerror: [m[K'[01m[Kstruct SGlobalState[m[K' has no member named '[01m[KtrailShader[m[K'
  253 |     glUniform4f(g_pGlobalState->[01;31m[KtrailShader[m[K.gradient, thisboxopengl.x, thisboxopengl.y, thisboxopengl.w, thisboxopengl.h);
      |                                 [01;31m[K^~~~~~~~~~~[m[K
[01m[K/build/hyprtrails/trail.cpp:254:33:[m[K [01;31m[Kerror: [m[K'[01m[Kstruct SGlobalState[m[K' has no member named '[01m[KtrailShader[m[K'
  254 |     glUniform4f(g_pGlobalState->[01;31m[KtrailShader[m[K.color, COLOR.r, COLOR.g, COLOR.b, COLOR.a);
      |                                 [01;31m[K^~~~~~~~~~~[m[K
[01m[K/build/hyprtrails/trail.cpp:260:43:[m[K [01;31m[Kerror: [m[K'[01m[Kstruct SGlobalState[m[K' has no member named '[01m[KtrailShader[m[K'
  260 |     glVertexAttribPointer(g_pGlobalState->[01;31m[KtrailShader[m[K.posAttrib, 2, GL_FLOAT, GL_FALSE, 0, (float*)points.data());
      |                                           [01;31m[K^~~~~~~~~~~[m[K
[01m[K/build/hyprtrails/trail.cpp:262:47:[m[K [01;31m[Kerror: [m[K'[01m[Kstruct SGlobalState[m[K' has no member named '[01m[KtrailShader[m[K'
  262 |     glEnableVertexAttribArray(g_pGlobalState->[01;31m[KtrailShader[m[K.posAttrib);
      |                                               [01;31m[K^~~~~~~~~~~[m[K
[01m[K/build/hyprtrails/trail.cpp:282:48:[m[K [01;31m[Kerror: [m[K'[01m[Kstruct SGlobalState[m[K' has no member named '[01m[KtrailShader[m[K'
  282 |     glDisableVertexAttribArray(g_pGlobalState->[01;31m[KtrailShader[m[K.posAttrib);
      |                                                [01;31m[K^~~~~~~~~~~[m[K
[01m[K/build/hyprtrails/trail.cpp:[m[K In member function '[01m[Kvirtual void CTrail::[01;32m[KdamageEntire[m[K()[m[K':
[01m[K/build/hyprtrails/trail.cpp:339:16:[m[K [01;35m[Kwarning: [m[Knarrowing conversion of '[01m[K(int)(((CTrail*)this)->CTrail::m_lastWindowPos.Hyprutils::Math::Vector2D::x - ((CTrail*)this)->CTrail::m_seExtents.Hyprutils::Math::SBoxExtents::topLeft.Hyprutils::Math::Vector2D::x)[m[K' from '[01m[Kint[m[K' to '[01m[Kdouble[m[K' [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/C_002b_002b-Dialect-Options.html#index-Wnarrowing-Wnarrowing]8;;[m[K]
  339 |     CBox dm = {[01;35m[K(int)(m_lastWindowPos.x - m_seExtents.topLeft.x)[m[K, (int)(m_lastWindowPos.y - m_seExtents.topLeft.y),
      |                [01;35m[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
[01m[K/build/hyprtrails/trail.cpp:339:66:[m[K [01;35m[Kwarning: [m[Knarrowing conversion of '[01m[K(int)(((CTrail*)this)->CTrail::m_lastWindowPos.Hyprutils::Math::Vector2D::y - ((CTrail*)this)->CTrail::m_seExtents.Hyprutils::Math::SBoxExtents::topLeft.Hyprutils::Math::Vector2D::y)[m[K' from '[01m[Kint[m[K' to '[01m[Kdouble[m[K' [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/C_002b_002b-Dialect-Options.html#index-Wnarrowing-Wnarrowing]8;;[m[K]
  339 |     CBox dm = {(int)(m_lastWindowPos.x - m_seExtents.topLeft.x), [01;35m[K(int)(m_lastWindowPos.y - m_seExtents.topLeft.y)[m[K,
      |                                                                  [01;35m[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
[01m[K/build/hyprtrails/trail.cpp:340:16:[m[K [01;35m[Kwarning: [m[Knarrowing conversion of '[01m[K(int)(((CTrail*)this)->CTrail::m_lastWindowSize.Hyprutils::Math::Vector2D::x + ((CTrail*)this)->CTrail::m_seExtents.Hyprutils::Math::SBoxExtents::topLeft.Hyprutils::Math::Vector2D::x + ((CTrail*)this)->CTrail::m_seExtents.Hyprutils::Math::SBoxExtents::bottomRight.Hyprutils::Math::Vector2D::x)[m[K' from '[01m[Kint[m[K' to '[01m[Kdouble[m[K' [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/C_002b_002b-Dialect-Options.html#index-Wnarrowing-Wnarrowing]8;;[m[K]
  340 |                [01;35m[K(int)(m_lastWindowSize.x + m_seExtents.topLeft.x + m_seExtents.bottomRight.x)[m[K, (int)m_seExtents.topLeft.y};
      |                [01;35m[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
[01m[K/build/hyprtrails/trail.cpp:340:95:[m[K [01;35m[Kwarning: [m[Knarrowing conversion of '[01m[K(int)((CTrail*)this)->CTrail::m_seExtents.Hyprutils::Math::SBoxExtents::topLeft.Hyprutils::Math::Vector2D::y[m[K' from '[01m[Kint[m[K' to '[01m[Kdouble[m[K' [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/C_002b_002b-Dialect-Options.html#index-Wnarrowing-Wnarrowing]8;;[m[K]
  340 |                (int)(m_lastWindowSize.x + m_seExtents.topLeft.x + m_seExtents.bottomRight.x), [01;35m[K(int)m_seExtents.topLeft.y[m[K};
      |                                                                                               [01;35m[K^~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
In file included from [01m[K/build/hyprtrails/main.cpp:13[m[K:
[01m[K/build/hyprtrails/globals.hpp:8:5:[m[K [01;31m[Kerror: [m[K'[01m[KCShader[m[K' does not name a type; did you mean '[01m[KSShader[m[K'?
    8 |     [01;31m[KCShader[m[K          trailShader;
      |     [01;31m[K^~~~~~~[m[K
      |     [32m[KSShader[m[K
[01m[K/build/hyprtrails/main.cpp:[m[K In function '[01m[Kvoid[01;32m[K initGlobal[m[K()[m[K':
[01m[K/build/hyprtrails/main.cpp:89:21:[m[K [01;31m[Kerror: [m[K'[01m[Kstruct SGlobalState[m[K' has no member named '[01m[KtrailShader[m[K'
   89 |     g_pGlobalState->[01;31m[KtrailShader[m[K.program   = prog;
      |                     [01;31m[K^~~~~~~~~~~[m[K
[01m[K/build/hyprtrails/main.cpp:90:21:[m[K [01;31m[Kerror: [m[K'[01m[Kstruct SGlobalState[m[K' has no member named '[01m[KtrailShader[m[K'
   90 |     g_pGlobalState->[01;31m[KtrailShader[m[K.proj      = glGetUniformLocation(prog, "proj");
      |                     [01;31m[K^~~~~~~~~~~[m[K
[01m[K/build/hyprtrails/main.cpp:91:21:[m[K [01;31m[Kerror: [m[K'[01m[Kstruct SGlobalState[m[K' has no member named '[01m[KtrailShader[m[K'
   91 |     g_pGlobalState->[01;31m[KtrailShader[m[K.tex       = glGetUniformLocation(prog, "tex");
      |                     [01;31m[K^~~~~~~~~~~[m[K
[01m[K/build/hyprtrails/main.cpp:92:21:[m[K [01;31m[Kerror: [m[K'[01m[Kstruct SGlobalState[m[K' has no member named '[01m[KtrailShader[m[K'
   92 |     g_pGlobalState->[01;31m[KtrailShader[m[K.color     = glGetUniformLocation(prog, "color");
      |                     [01;31m[K^~~~~~~~~~~[m[K
[01m[K/build/hyprtrails/main.cpp:93:21:[m[K [01;31m[Kerror: [m[K'[01m[Kstruct SGlobalState[m[K' has no member named '[01m[KtrailShader[m[K'
   93 |     g_pGlobalState->[01;31m[KtrailShader[m[K.texAttrib = glGetAttribLocation(prog, "colors");
      |                     [01;31m[K^~~~~~~~~~~[m[K
[01m[K/build/hyprtrails/main.cpp:94:21:[m[K [01;31m[Kerror: [m[K'[01m[Kstruct SGlobalState[m[K' has no member named '[01m[KtrailShader[m[K'
   94 |     g_pGlobalState->[01;31m[KtrailShader[m[K.posAttrib = glGetAttribLocation(prog, "pos");
      |                     [01;31m[K^~~~~~~~~~~[m[K
[01m[K/build/hyprtrails/main.cpp:95:21:[m[K [01;31m[Kerror: [m[K'[01m[Kstruct SGlobalState[m[K' has no member named '[01m[KtrailShader[m[K'
   95 |     g_pGlobalState->[01;31m[KtrailShader[m[K.gradient  = glGetUniformLocation(prog, "snapshots");
      |                     [01;31m[K^~~~~~~~~~~[m[K
make[2]: *** [CMakeFiles/hyprtrails.dir/build.make:121: CMakeFiles/hyprtrails.dir/trail.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [CMakeFiles/hyprtrails.dir/build.make:107: CMakeFiles/hyprtrails.dir/main.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:87: CMakeFiles/hyprtrails.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

### Additional context

_No response_

### System metadata

- system: `"x86_64-linux"`
 - host os: `Linux 6.14.6, NixOS, 24.11 (Vicuna), 24.11.20250515.ba8b70e`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.14`
 - nixpkgs: `/nix/store/50w2kcwpvs2p0n4waldx0ci9y6qdig27-source`

(Since the system is not building because of this issue, it still shows 24.11, where it should be 25.05)

### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/window-managers/hyprwm/hyprland-plugins/hyprland-plugins.nix
+++ b/pkgs/applications/window-managers/hyprwm/hyprland-plugins/hyprland-plugins.nix
@@ -14,13 +14,13 @@ let
             mkHyprlandPlugin,
           }:
           let
-            version = "0.49.0";
+            version = "0.49.0-fix";
 
             hyprland-plugins-src = fetchFromGitHub {
               owner = "hyprwm";
               repo = "hyprland-plugins";
               tag = "v${version}";
-              hash = "sha256-GpsLyK/U05q7QnyFIWrnGS2loVyjPZByTtPitwu9UNw=";
+              hash = "sha256-5wjvmFtk85uBJLnrKZbfrgo9OTTQlFP18ZNgfBwFsz0=";
             };
           in
           mkHyprlandPlugin hyprland {

